x-pack/plugins/rollup/public/crud_app/services/api.js

```diff
@@ -30,6 +31,8 @@ export async function deleteJobs(jobIds) {
 }
 
 export async function createJob(job) {
+  // Note that this will record an acion, even if the related request fails.
+  trackUserAction('create-rollup-job');
```

Comment:
Note for reviewers: I think we should track all user actions by hitting this endpoint from the client, and not do any "proxied" user action tracking within CRUD API endpoints. I think this will a) make it easier to keep track of which actions are being tracked, and b) add greater fidelity to our data in the event that our API endpoints are consumed by other apps or third parties (e.g. command line or an automated tool). What does everyone else think?

Reply:
From a telemetry data perspective, why do we care if people created a rollup job by clicking a button in the UI versus using the API that powers that button? The benefit of putting the telemetry at the source of truth (the server) is that it represents all usage of the feature.

Reply:
To provide more context, some of the actions we want to track aren't proxied by an API endpoint, e.g. clicking a particular tab or navigating somewhere. So I think the main benefit of keeping all tracking client-side will be that it's easier to keep track of what's being tracked, though maybe I'm overrating that benefit.

Reply:
I see.